### ISSUE TITLE
Set config values via UI.

### ISSUE THREAD

**Issue by tillkruss** [tillkruss/redis-cache#173]
After #140 is implemented, maybe we can build a `wp-config.php` setter that allows users to easily configure the plugin.

**Comment by naxvog** [comment]
Rather than editing the `wp-config.php` itself I would suggest building a config file for the plugin to be included every time configuration constants are not set to allow for backwards compatibility.

The configuration would be written to an own directory (let's say `/wp-content/redis-cache-config`) using a sample-config file. I have implemented such functionality for my WPR-Redis Plugin (shameless plug):
* [Configuration template file](https://github.com/naxvog/wpr-redis/blob/8a78d814887aa10e164f1cce9c28836b7cce1425/includes/config-template.php) as placeholder for the configuration to write into.
* [Configuration values are still written to the options table](https://github.com/naxvog/wpr-redis/blob/fcb8145228eb992145115ebc8f675c895c94e1ba/includes/class-config.php#L120) utilizing the [WordPress Settings API](https://codex.wordpress.org/Settings_API)
* [Save method](https://github.com/naxvog/wpr-redis/blob/fcb8145228eb992145115ebc8f675c895c94e1ba/includes/class-config.php#L327) writing the saved values to the configuration file. [Triggering upon setting update](https://github.com/naxvog/wpr-redis/blob/fcb8145228eb992145115ebc8f675c895c94e1ba/includes/class-config.php#L279) for maximum compatibility. 

The WP-Rocket plugin uses such functionality to save its configuration too. This method assures that `wp-config.php` is not polluted by constants that the normal user is not recognizing and circumvents any write errors to the WordPress configuration that would most certainly result in a WSOD.

**Comment by tillkruss** [comment]
Aha, so load the custom config file as part of `object-cache.php` with defaults?

**Comment by naxvog** [comment]
We should check if configuration constants were set before for example in an APF file or as part of the `wp-config.php` and only load defaults if those were not found. But yes the config should be required in the `object-cache.php` file.

**Comment by tillkruss** [comment]
**Basics**

- Client
- Host
- Port
- Password
- Database
- Prefix/Salt

**Nice to have**

- MaxTTL
- Timeout + Read Timeout
- Selective flushing 
- Ignored groups
- Serializer
- igbinary

**Comment by tullibo** [comment]
+1 for this dude, we're using on shared environments in Cloudways and Plesk and the cache is hopping between sites. Would be nice if the cache salt was taken care of automatically or in the gui

**Comment by brunobertechini** [comment]
Don't forget the default for TLS as well (in my case we have a default redis pointing to Azure Cache with SSL always on)
